src/analysis/exploration/biomassPrecursorCheck.m

```diff
@@ -24,20 +36,44 @@
 % AND OPTIMIZE.  NOTE: A CRITICAL ASSUMPTION IS THAT THE ADDED DEMAND
 % REACTION IS APPENDED TO THE FAR RIGHT OF THE S-MATRIX.  THE CODE NEEDS TO
 % BE REVISED IF THIS IS NOT THE CASE.
-k = 1;
 m = 1;
+p = 1;
+c = 1;
 % ADD DEMAND REACTIONS
 [model_newDemand, addedRxns] = addDemandReaction(model, biomassMetabs);
+
+if checkCoupling
+    %Close the precursors
+    model_newDemand = changeRxnBounds(model_newDemand,addedRxns,zeros(numel(addedRxns,1)),repmat('b',numel(addedRxns,1)));
+    coupledMets = {};
+end
+
 [missingMets, presentMets] = deal({});
 for i = 1:length(biomassMetabs)
+    if checkCoupling
+        model_newDemand = changeRxnBounds(model_newDemand, addedRxns{i}, 1000, 'u');
+    end
     model_newDemand = changeObjective(model_newDemand, addedRxns{i});
     solution = optimizeCbModel(model_newDemand);                                % OPTIMIZE
     if solution.f == 0                                                          % MAKE LIST OF WHICH BIOMASS PRECURSORS ARE ...
-        missingMets(k) = biomassMetabs(i);                                      %  SYNTHESIZED AND WHICH ARE NOT
-        k = k + 1;
+        if checkCoupling
+            model_newDemand = changeRxnBounds(model_newDemand, addedRxns, 1000, 'u');
+            solution = optimizeCbModel(model_newDemand);
+            if solution.f > 0
+                coupledMets(c) = biomassMetabs(i);                              % NEED ANOTHER SINK

```

Comment:
Why are all these comments in uppercase letters?

Reply:
Don't ask me. thats the original code....

Reply:
Hm. Would be nice to adapt them to lowercase.